devito/interfaces.py

```diff
@@ -1,29 +1,102 @@
 import numpy as np
 from sympy import IndexedBase
 from tools import aligned
-
+import os
+from signal import *
+import sys
 
 __all__ = ['DenseData', 'TimeData', 'PointData']
 
 
 class DenseData(IndexedBase):
-    def __init__(self, name, shape, dtype):
+
+    def __init__(self, name, shape, dtype, disk_path=None):
         self.name = name
         self.dtype = dtype
         self.pointer = None
         self.initializer = None
+        self.disk_path = disk_path
         super(DenseData, self).__init__(name)
 
     def __new__(cls, *args, **kwargs):
         return IndexedBase.__new__(cls, args[0], shape=args[1])
 
+    # this is the default directory where memmap files will be created
+    _default_disk_path = None
+    # holds the string name of all memmap file created
+    _memmap_file_list = []
+    # default exit code used on interrupt and kill
+    _default_exit_code = 0
+
+    # functions for managing memmap files
+    # call this method to specify where you want memmap file to be created
+    @staticmethod
+    def set_default_disk_path(disk_path):
```

Comment:
The way I see it you are re-using the disk_path variable as a flag for whether to use memmap or not. First thing to do here is to make the API clear and have a separate flag for whether to use memmap or not. We can not expect the user (application developer) to understand that passing a path will change the memory behaviour. This, I believe, is absolutely essential.

For the rest of it, here is my proposed behaviour. Let me know what you think:
- Memmap is disabled by default in ALL cases. It has to be enabled on a per object basis regardless of what some global variables might look like. We have to have consistent behaviour on object constructors and not depend on the global state.
- When enabled, we look at whether `disk_path` is set. If it is, we use that.
- If it is not, we ask the OS for a temp path, the way we do in the JIT section.